wgsl/index.bs

```diff
@@ -8975,6 +8975,9 @@ Each is described in detail in subsequent sections.
       Sample index for the current fragment.  The value is least 0 and at most
       `sampleCount`-1, where `sampleCount` is the MSAA sample
       {{GPUMultisampleState/count}} specified for the GPU render pipeline.
+      When this attribute is applied, if the effects of the fragment shader would vary based
```

Comment:
This sounds like a constraint that if the effects of the FS do not depend on the sample_index, then the fragment shader must be executed only once. Does storage buffer writes count as effects? If yes, I believe it is non-observable whether the FS is always executed 4 times, or only when there are side effects (aside from pipeline statistics queries, which we don't support)

Reply:
I don't know what to put there because I don't understand what Metal is trying to say. Metal has wording that seem to effectively say "might do per-sample" whereas Vulkan, D3D and OpenGL just say "will do per-sample". My interpretation (guess?) was that Metal want to say "will do per-sample if observable" which seems a little silly to even bring up, if I can't observe it why would I care it did something more optimal? Metal's wording is more like "we might try to do the minimum work so, if we can separate the shader into parts that are affected by sample_index and parts that are not, we might run the parts that are not affected, per-pixel instead of per-sample.

> Does storage buffer writes count as effects? If yes, I believe it is non-observable whether the FS is always executed 4 times

```
atomicAdd(&numTimesFSWasExecuted, 1);
```

will let you observe how many times the FS is executed